pyrate/core/dem_error.py

```diff
@@ -14,299 +14,285 @@
+                lat_parts = lat[t.top_left_y:t.bottom_right_y, t.top_left_x:t.bottom_right_x]
+                az_parts = az[t.top_left_y:t.bottom_right_y, t.top_left_x:t.bottom_right_x]
+                rg_parts = rg[t.top_left_y:t.bottom_right_y, t.top_left_x:t.bottom_right_x]
+
+                nifgs = len(ifg_paths)
+                bperp = np.empty((nifgs, lon_parts.shape[0], lon_parts.shape[1])) * np.nan
+                ifg_num = 0
+                # calculate per-pixel perpendicular baseline for each IFG
+                for ifg_path in ifg_paths:  # loop could be avoided by approximating the look angle for the first Ifg
+                    ifg = Ifg(ifg_path)
+                    ifg.open(readonly=True)
+                    # calculate look angle for interferograms (using the Near Range of the primary SLC)
+                    look_angle, range_dist = geometry.calc_local_geometry(ifg, None, rg_parts, lon_parts, lat_parts,
+                                                                          params)
+                    bperp[ifg_num, :, :] = geometry.calc_local_baseline(ifg, az_parts, look_angle)
+                    ifg_num += 1
+
+                log.debug('Calculating DEM error for tile {} during DEM error correction'.format(t.index))
+                # mst_tile = np.load(Configuration.mst_path(params, t.index))
+                # calculate the DEM error estimate and the correction values for each IFG
+                # current implementation uses the look angle and range distance matrix of the primary SLC in the last IFG
+                # todo: check the impact of using the same information from another SLC
+                dem_error, dem_error_correction = calc_dem_errors(ifg_parts, bperp, look_angle, range_dist, threshold)
+                # dem_error contains the estimated DEM error for each pixel (i.e. the topographic change relative to the DEM)
+                # size [row, col]
+                # dem_error_correction contains the correction value for each interferogram
+                # size [num_ifg, row, col]
+
+                # save tiled data in tmpdir
+                np.save(file=os.path.join(params[cf.TMPDIR], 'dem_error_{}.npy'.format(t.index)), arr=dem_error)
+                # swap the axes of 3D array to fit the style used in function assemble_tiles
+                tmp_array = np.moveaxis(dem_error_correction, 0, -1)
+                # new dimension is [row, col, num_ifg]
+                # save tiled data into tmpdir
+                np.save(file=os.path.join(params[cf.TMPDIR], 'dem_error_correction_{}.npy'.format(t.index)), \
+                        arr=tmp_array)
+
+            # wait for all processes to finish
+            mpiops.comm.barrier()
+
+            # write dem error and correction values to file
+            mpiops.run_once(_write_dem_errors, ifg_paths, params, preread_ifgs, tiles)
+            shared.save_numpy_phase(ifg_paths, params)
+
```

Comment:
It's possbile to use the `tiles_split` function here. In order to do that you may need to look into tiled or IfgPart equivalent of the `Geometry` class.